compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5266,37 +5266,10 @@ TR::Register *OMR::Power::TreeEvaluator::loadaddrEvaluator(TR::Node *node, TR::C
       else
          {
          int64_t  offset = mref->getOffset(*comp);
-         if (mref->hasDelayedOffset() || offset!=0 || comp->getOption(TR_AOT))
+         if (mref->hasDelayedOffset() || !mref->getBaseRegister() || offset!=0 || comp->getOption(TR_AOT))
             {
```

Comment:
checking getBaseRegister()==NULL or explicitly checking mref->getLabel() != NULL  is more correct?  which situation you actually intended to?

Reply:
This is actually specifically checking for either the case where we're performing an `addi rX, 0(r0)` _or_ when we're performing a PC-relative load. The else branch here assumes that the base register can be reused as the register for this node, which obviously doesn't work if there is no base register.

To make this clearer, I've gone ahead and switched the code to check both conditions explicitly.